main/src/library/MutList.flix

```diff
@@ -716,7 +716,7 @@ namespace MutList {
     ///
     /// Apply the effectful function `f` to all the elements in `v`.
     ///
-    pub def foreach(f: a ~> Unit, v: MutList[a]): Unit & Impure =
+    pub def foreach(f: a -> Unit & ef, v: MutList[a]): Unit & ef =
```

Comment:
f should have effect `ef` but the whole function should be impure (because it touches mutable memory.)